inference-engine/src/low_precision_transformations/src/split.cpp

```diff
@@ -22,118 +23,69 @@ bool SplitTransformation::transform(TransformationContext& context, ngraph::patt
         return false;
     }
 
-    const std::shared_ptr<Node> split = NetworkHelper::separateInStandaloneBranch(m.get_match_root());
-    auto dequantization = NetworkHelper::getDequantization(split);
+    const auto split = NetworkHelper::separateInStandaloneBranch(m.get_match_root());
+    const auto dequantization = NetworkHelper::getDequantization(split);
 
-    OutputVector inputs(split->get_input_size());
-    for (size_t i = 0; i < split->get_input_size(); ++i) {
-        inputs[i] = split->get_input_node_shared_ptr(i);
-    }
-
-    const size_t dequantizationIndex = NetworkHelper::getChildInputIndex(dequantization.multiply, split);
-    inputs[dequantizationIndex] = dequantization.data;
+    OutputVector inputs = split->input_values();
+    inputs[0] = dequantization.data;
 
-    std::shared_ptr<ngraph::Node> newSplit = split->clone_with_new_inputs(inputs);
+    const auto newSplit = split->clone_with_new_inputs(inputs);
     newSplit->set_friendly_name(split->get_friendly_name());
+    ngraph::copy_runtime_info(split, newSplit);
 
-    const ngraph::Shape subConstShape = dequantization.subtract ?
-        dequantization.subtract->get_input_node_shared_ptr(1)->get_shape() : Shape{};
-    std::vector<float> subValues = dequantization.subtract ? as_type_ptr<opset1::Constant>(
-        dequantization.subtract->get_input_node_shared_ptr(1))->cast_vector<float>() : std::vector<float>();
-
-    const ngraph::Shape mulConstShape = dequantization.multiply->get_input_node_shared_ptr(1)->get_shape();
-    std::vector<float> mulValues = as_type_ptr<opset1::Constant>(
-        dequantization.multiply->get_input_node_shared_ptr(1))->cast_vector<float>();
-
-    int64_t SplitedAxis = as_type_ptr<opset1::Constant>(split->get_input_node_shared_ptr(1))->cast_vector<int64_t>()[0];
-    size_t axis = SplitedAxis > 0 ? SplitedAxis : split->get_input_shape(0).size() + SplitedAxis;
-    size_t outputSize = newSplit->get_output_size();
-
-    const auto subSplitLengths = getConstSplitLengths(inputs, subConstShape, outputSize);
-    const auto mulSplitLengths = getConstSplitLengths(inputs, mulConstShape, outputSize);
+    int64_t axis = as_type_ptr<opset1::Constant>(split->get_input_node_shared_ptr(1))->cast_vector<int64_t>()[0];
+    size_t normalizeAxis = normalize_axis(split->get_friendly_name(), axis, split->get_input_partial_shape(0).rank());
 
-    std::vector<std::shared_ptr<ngraph::Node>> lastNodes(outputSize);
-    ngraph::OutputVector replacement;
-    for (size_t i = 0; i < outputSize; ++i) {
+    NodeVector lastNodes;
+    OutputVector replacement;
+    for (size_t i = 0; i < newSplit->get_output_size(); ++i) {
```

Comment:
I think to iterate for each output and fold `Split`/`VariadicSplit` for each output for the same dequantization operation repeatedly can be simpler for understanding, but I would like to decrease `fold` operations. Can you, please, refactor: fold `Split`/`VariadicSplit` only twice (`Subtract` and `Multiply` constants) and handle fold results for each output? Thanks!

Reply:
Done, thanks!